pkgs/servers/sql/postgresql/ext/omnigres.nix

```diff
@@ -0,0 +1,118 @@
+{
+  buildEnv,
+  bison,
+  boost,
+  brotli,
+  bzip2,
+  cmake,
+  doxygen,
+  fetchFromGitHub,
+  flex,
+  git,
+  lib,
+  libcap,
+  libtool,
+  libuv,
+  libxml2,
+  lld,
+  netcat,
+  openssl,
+  perl,
+  pkg-config,
+  postgresql,
+  postgresqlBuildExtension,
+  postgresqlTestExtension,
+  postgresqlPackages,
+  python3,
+  readline,
+  stdenv,
+  zlib,
+}:
+
+let
+  # TODO: I'll remove this as well, just testing stuff
+  extensions =
+    buildEnv {
+      name = "postgresql-custom";
+      paths = [ postgresql postgresqlPackages.plpython3 ];
+    };
+in
+postgresqlBuildExtension (finalAttrs: {
+  pname = "omnigres";
+  version = "74ab3167d32e47fec790bb54a979780f7f9f1a28";
+
+  src = fetchFromGitHub {
+    owner = "omnigres";
+    repo = "omnigres";
+    rev = finalAttrs.version;
+    hash = "sha256-PuxfD4XJv6ZxD1LnF3OopY+m057eBT1ZvEfRSbbIyzQ=";
+  };
+
+  nativeBuildInputs = [ cmake flex libtool pkg-config ];
+  buildInputs =
+    postgresql.buildInputs ++ [
+      bison
+      boost
+      brotli
+      bzip2
+      doxygen
+      flex
+      git
+      libcap
+      libuv
+      libxml2
+      lld
+      netcat
+      openssl
+      perl
+      python3
+      readline
+      zlib
+      # TODO: just some hacky tests, remove this later
+      extensions
+    ];
+
+  preConfigure = ''
+    substituteInPlace cmake/PostgreSQLExtension.cmake \
+      --replace "\''${_ext_dir}" "$out"
+  '';
```

Comment:
@yrashk Can you confirm if this path does make any sense?

Reply:
The intention there is that this is only necessary if in-tree builds of Postgres are used, which is not the case with Nix.

So ultimately it shouldn't necessary to do this as long as `install_extensions` or `install_EXT_extension` targets are called, they should install them to wherever pg_config points them to. If it insists on this, it might be something on our end to fix to facilitate.

Reply:
Without this substitution I get into issues during `copy` calls

https://github.com/omnigres/omnigres/blob/master/cmake/PostgreSQLExtension.cmake#L273